### PR TITLE
Bump Deplyment Target to iOS 15

### PR DESCRIPTION
**Pull request by zeitschlag**

### CODE REVIEW

File: Mastodon/Extension/UICollectionViewDiffableDataSource.swift [hunk 1]
```diff
@@ -13,28 +13,14 @@ extension UICollectionViewDiffableDataSource {
...
     ) {
-        if #available(iOS 15.0, *) {
-            self.applySnapshotUsingReloadData(snapshot, completion: completion)
-        } else {
-            self.apply(snapshot, animatingDifferences: false, completion: completion)
-        }
+        self.applySnapshotUsingReloadData(snapshot, completion: completion)
     }
     
     func applySnapshot(
```
Comment (j-f1) [inline Mastodon/Extension/UICollectionViewDiffableDataSource.swift]: I think these functions can be dropped entirely in favor of just calling `apply(snapshot, ...)`, right?

File: MastodonSDK/Sources/MastodonUI/Vendor/MetaTextView+PasteExtensions.swift [hunk 2]
```diff
@@ -15,19 +15,17 @@ extension MetaTextView {
...
-            var nextResponder = self.next;
-            
-            // Force the event to bubble through ALL responders
-            // This is a workaround as somewhere down the chain the paste event gets eaten
-            while (nextResponder != nil) {
-                if let nextResponder = nextResponder {
-                    if (nextResponder.responds(to: #selector(UIResponderStandardEditActions.paste(_:)))) {
-                        nextResponder.perform(#selector(UIResponderStandardEditActions.paste(_:)), with: sender)
-                    }
+        var nextResponder = self.next;
```
Comment (j-f1) [inline MastodonSDK/Sources/MastodonUI/Vendor/MetaTextView+PasteExtensions.swift:18]: This code could be rewritten using [`sequence(first:next:)`](https://developer.apple.com/documentation/swift/sequence(first:next:)) (although I’m not sure if it needs to be)

```swift
for responder in sequence(first: self, next: \.next) {
    if responder.responds(to: #selector(UIResponderStandardEditActions.paste(_:))) {
        responder.perform(#selector(UIResponderStandardEditActions.paste(_:)), with: sender)
    }
}
```

**Comment by j-f1** [comment]
Also, are you working on handling the deprecation warnings? I can take a look at those if you’re not!

File: Mastodon/Extension/UICollectionViewDiffableDataSource.swift (hunk 1 above)
Comment (zeitschlag) [reply Mastodon/Extension/UICollectionViewDiffableDataSource.swift]: You are absolutely right 👍 
Great, that you saw that :)

**Comment by zeitschlag** [comment]
@j-f1 Good point. If you see something, feel free to fix it :) I'll do so myself, too. Thanks for you help!

From what I see, most warnings are `'contentEdgeInsets' was deprecated in iOS 15.0: This property is ignored when using UIButtonConfiguration`, but we don't use configurations (yet)

**Review by j-f1 — APPROVED** [review]
LGTM!

**Review by sladewatkins — APPROVED** [review]
LGTM as well!

**Review by kimar — APPROVED** [review]
LGTM!

**Pull request merged**